pkg/util/shell/shell_test.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shell
+
+import (
+	"os"
+	"testing"
+)
+
+type shellTestCase struct {
+	name         string
+	cmdLine      string
+	expectedHint string
+}
+
+type prefixSuffixDelimiterTestCase struct {
+	shellName string
+	prefix    string
+	suffix    string
+	pathVar   bool
+	delimiter string
+}
+
+func TestUnknownShell(t *testing.T) {
+	expectedShellName := "foo"
+	defer func(shell string) { os.Setenv("SHELL", shell) }(os.Getenv("SHELL"))
+	os.Setenv("SHELL", "/bin/"+expectedShellName)
+
+	shell, err := GetShell("")
+	if err != nil {
+		t.Fatal("Unexpected err : ", err.Error())
+	}
+
+	if shell != expectedShellName {
+		t.Fatalf("Expected shell to be %s but %s", expectedShellName, shell)
+	}
+}
+
+func TestKnownShell(t *testing.T) {
+	expectedShellName := "bash"
+	shell, err := GetShell("bash")
```

Comment:
@hferentschik , the above one works as `GenerateUsageHint` accepts any shell name and if it is not "fish", or some defined shell, it default to bash like hint.

```
$ minishift oc-env --shell /bin/fish
export PATH="/home/budhram/.minishift/cache/oc/v1.5.1:$PATH"
# Run this command to configure your shell:
# eval $(minishift oc-env)

$ minishift oc-env --shell fish
set -gx PATH "/home/budhram/.minishift/cache/oc/v1.5.1" $PATH;
# Run this command to configure your shell:
# eval (minishift oc-env)
```